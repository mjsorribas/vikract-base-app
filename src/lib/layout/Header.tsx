import { Box, Flex } from "@chakra-ui/react";

const Header = () => {
  return (
    <Flex
      as="header"
      width="full"
      align="center"
      alignSelf="flex-start"
      justifyContent="center"
      gridGap={2}
    >
      <Box marginLeft="auto" />
    </Flex>
  );
};

export default Header;
