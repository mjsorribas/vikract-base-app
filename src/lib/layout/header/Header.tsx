import { Box } from "@chakra-ui/react";

import { theme } from "lib/styles/customTheme";

import Navbar from "./components/navbar/NavBar";

const Header = () => {
  return (
    <Box
      w="100%"
      padding="8"
      bg={theme.colors.cream}
      color={theme.colors.black}
      maxHeight="100px"
      as="header"
    >
      <Navbar />
    </Box>
  );
};

export default Header;
