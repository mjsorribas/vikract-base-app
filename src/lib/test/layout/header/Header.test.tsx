import { ChakraProvider } from "@chakra-ui/react"; // Importa ChakraProvider
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Header from "lib/layout/header/Header";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Header", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <ChakraProvider>
        {" "}
        {/* Agrega ChakraProvider aqu√≠ */}
        <Header />
      </ChakraProvider>
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
