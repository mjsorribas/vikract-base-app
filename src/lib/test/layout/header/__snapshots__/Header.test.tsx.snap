// Vitest Snapshot v1

exports[`Header 1`] = `
<header
  className="css-13negw7"
>
  <div
    className="css-0"
  >
    <div
      className="css-k008qs"
    >
      <div
        className="css-70qvj9"
      >
        <div
          className="css-0"
        >
          <img
            className="chakra-image css-1chmwhm"
            src="/assets/vikract.svg"
            title="The Vikract Webapp template"
          />
          <p
            className="chakra-text css-104caxi"
          >
            THE VIKRACT TEMPLATE
          </p>
        </div>
      </div>
      <div
        className="css-17xejub"
      />
      <div
        className="css-70qvj9"
      >
        <div
          className="chakra-stack css-ex90l4"
        >
          <a
            className="chakra-link css-177s7ri"
            href="/"
          >
            Home
          </a>
          <a
            className="chakra-link css-1jpe0vt"
            href="/prices"
          >
            Prices
          </a>
          <a
            className="chakra-link css-1jpe0vt"
            href="/contactus"
          >
            Contact Us
          </a>
        </div>
      </div>
      <div
        className="css-17xejub"
      />
      <div
        className="css-0"
      >
        <div
          className="css-0"
        >
          <div
            className="chakra-stack css-111rh5x"
          >
            <button
              className="chakra-button css-4xx2wk"
              type="button"
            >
              Sign Up
            </button>
            <button
              className="chakra-button css-4xx2wk"
              type="button"
            >
              Sign In
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
